### PR TITLE
Tic tac toe

### PR DESCRIPTION
**Pull request by DavidRojas1612**
## Summary
- Completed exercise tic-tac-toe from reactjs.org documentation 

## Screenshots
image

## Trello Card
https://trello.com/c/8PICGUXO

### CODE REVIEW

File: react/tic-tac-toe/src/app/screens/Game/index.js [hunk 1]
```diff
@@ -1,20 +1,65 @@
...
+    const status = winner ? `Winner: ${winner}` : `Next player: ${xIsNext ? 'X' : 'O'}`;
+
+    return (
+      <div className={styles.game}>
+        <div className={styles.gameBoard}>
+          <Board squares={current.squares} onClick={this.handleClick} />
+        </div>
+        <div className={styles.gameInfo}>
+          <div>{status}</div>
+          <ol>{this.handleViewMoves(history)}</ol>
```
Comment (jmtov) [inline react/tic-tac-toe/src/app/screens/Game/index.js]: Move the mapping here instead, would be clearer to understand what's going on and might actually not make a huge difference in performance.

File: react/tic-tac-toe/src/app/screens/Game/components/Moves/index.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+
+function Moves({ onClick, move }) {
+  const desc = move ? `Go to move #${move}` : 'Go to game start';
+  const handleClick = useCallback(() => {
```
Comment (alejobh) [inline react/tic-tac-toe/src/app/screens/Game/components/Moves/index.js]: i really appreciate this way!! simply nice

File: react/tic-tac-toe/src/app/screens/Game/components/Moves/index.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    onClick(move);
+  }, [move]);
+
+  return (
+    <li key={move}>
+      <button type="button" onClick={hanldeClick}>
```
Comment (alejobh) [inline react/tic-tac-toe/src/app/screens/Game/components/Moves/index.js]: BUT, `handleClick` ?

File: react/tic-tac-toe/src/app/screens/Game/components/Square/index.js [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
-    {/* TODO */}
-  </button>
-);
+function Square({ onClick, position, value }) {
+  const handleClick = useCallback(() => {
+    onClick(position);
+  }, [position]);
+
+  return (
+    <button type="button" className={styles.square} onClick={handleOnclick}>
```
Comment (alejobh) [inline react/tic-tac-toe/src/app/screens/Game/components/Square/index.js]: `handleClick`

File: react/tic-tac-toe/src/app/screens/Game/components/Square/index.js [hunk 5]
```diff
@@ -1,11 +1,24 @@
...
+      {value}
+    </button>
+  );
+}
+
+Square.propTypes = {
+  onClick: PropTypes.func.isRequired,
+  position: PropTypes.number,
```
Comment (alejobh) [inline react/tic-tac-toe/src/app/screens/Game/components/Square/index.js]: is this `required` ?

File: react/tic-tac-toe/src/utils/index.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+function calculateWinner(squares) {
```
Comment (alejobh) [inline react/tic-tac-toe/src/utils/index.js]: use arrow functions for utils

**Pull request merged**